libs/framework/include/celix/dm/ServiceDependency_Impl.h

```diff
@@ -22,11 +22,26 @@
 #include <cstring>
 #include "celix_constants.h"
 #include "celix_properties.h"
+#include "ServiceDependency.h"
+
 
 using namespace celix::dm;
 
+inline void BaseServiceDependency::runBuild() {
+    bool alreadyAdded = depAddedToCmp.exchange(true);
+    if (!alreadyAdded) {
+        celix_dmComponent_addServiceDependency(cCmp, cServiceDep);
+    }
+}
+
+inline BaseServiceDependency::~BaseServiceDependency() noexcept {
+    if (!depAddedToCmp) {
```

Comment:
Can you double check if the `!` is correct here?

Reply:
This is correct. If a dependency is added to a dm component it will be destroyed when the component is destroyed. 
If a dependency is not added to a dm component (i.e. not build) it needs to be destroyed explicitly.